tox.ini

```diff
@@ -1,12 +1,24 @@
 [tox]
-envlist = py37,flake8
+# Use a new pip so we can use PEP 508 URL requirements
+requires = pip >= 19.1.1
```

Comment:
Are you sure about this? [Release notes of pip 19.1.1](https://pip.pypa.io/en/stable/news/#id1) doesn't mention anything about [PEP 508 URL requirements feature](https://www.python.org/dev/peps/pep-0508/#examples). Also I don't see you to use this feature in your pull request, looking at `git blame` it was added by Coady.

Pip 19.1.1 is very recent (2019-05-06), so that it isn't even packaged in Fedora 30.  When I tried it with `python3-pip-19.0.3-1.fc30.noarch`, it worked just fine. So I would rather not have the minimal version high for no reason. 

I suggest to either drop minimal version or find which pip version actually introduced the feature you are referencing here.

Reply:
@mbukatov when I attempted to use the url requirement pip told me I needed to update. I wasn't able to find the specific release version that included that PEP so I just went with the latest. I just tested with a fresh venv and it seems like installing `19.0` works but `18.0` doesn't. I think as long as we require `pip>=19` we should be good to go on this one.

Reply:
Ah, ok. I checked the release notes and see that the feature was added in 18.1 (2018-10-05):

> Allow PEP 508 URL requirements to be used as dependencies.

So let's use that.

Reply:
We need to specify a version [`>= 18.1`](https://pip.pypa.io/en/stable/news/#id90) for the PEP 508 feature. You're correct that I didn't add the usage of it, but tests in TravisCI with python3.6 will fail without a version that's new enough.

Re: Fedora - these dependencies will be installed by pip, and not the package manager.
Re: the version being recent - is there a reason to not use the latest version? I guess I can back it down to 18.1 if you really think it matters

Reply:
> Re: Fedora - these dependencies will be installed by pip, and not the package manager.
> Re: the version being recent - is there a reason to not use the latest version? I guess I can back it down to 18.1 if you really think it matters

When running tests with tox, you are correct that the actual test environment is installed via pip into temporary virtualenv, which is isolated from whatever is packaged in the distribution, and for this reason it doesn't really matter which version we pick. But this doesn't apply for the python, pip, virtualenv and tox itself, which needs to be already installed before running the tests.

So I believe that the best approach here is to make sure that the tests will work with tox as packaged in Fedora (or any other recent distro), so that when one has tox installed from the distribution package, the tests will just work.

If we require the latest pip or tox, one needs to install new pip over the one from the distribution, which could create additional issues.

I noted this in the [README file for the unit tests](https://github.com/red-hat-storage/ocs-ci/blob/master/docs/unit_tests.md).

Reply:
Testing with the latest Fedora is fine, but we should also be testing with RHEL8 IMO. The reason we need to update pip, though is that Travis builds failing without doing that. It also doesn't need to install over the system-provided copy - it does that in a virtualenv, like everything else tox does.

Reply:
@zmc see https://github.com/red-hat-storage/ocs-ci/pull/254